cadc-util/src/main/java/ca/nrc/cadc/auth/AuthenticationUtil.java

```diff
@@ -138,76 +134,70 @@ public class AuthenticationUtil {
-        String cname = System.getProperty(Authenticator.class.getName());
-        Class c = null;
-        if (cname == null) {
-            cname = defaultImplClass;
-        }
-        try {
-            c = Class.forName(cname);
-            Object o = c.newInstance();
-            Authenticator ret = (Authenticator) o;
-            log.debug("Authenticator: " + cname);
-            return ret;
-        } catch (Throwable t) {
-            if (!defaultImplClass.equals(cname) || c != null) {
-                log.error("failed to load Authenticator: " + cname, t);
-            }
-            log.debug("failed to load Authenticator: " + cname, t);
-        }
-        log.debug("Authenticator: null");
-        return null;
-    }
-
     /**
      * Load the available IdentityManager implementation. This utility method will
      * check the <code>ca.nrc.cadc.auth.IdentityManager</code> system property for a
-     * configured class name (default class name:
-     * <code>ca.nrc.cadc.auth.IdentityManagerImpl</code>). The easiest way for
-     * software implementers to customize behavior is to create their own
-     * IdentityManagerimpl class and add it to the classpath.
+     * configured class name. If not configured or loading the configured implementation
+     * fails, the default is the <code>ca.nrc.cadc.auth.NoOpIdentityManager</code>.
      * 
-     * @return an IdentityManager implementation or null if none provided
+     * @return an IdentityManager implementation
      */
     public static IdentityManager getIdentityManager() {
-        String defaultImplClass = IdentityManager.class.getName() + "Impl";
         String cname = System.getProperty(IdentityManager.class.getName());
-        Class c = null;
-        if (cname == null) {
-            cname = defaultImplClass;
+        if (cname != null) {
+            try {
+                Class c = Class.forName(cname);
+                Object o = c.getConstructor().newInstance();
+                IdentityManager ret = (IdentityManager) o;
+                log.debug("IdentityManager: " + cname);
+                return ret;
+            } catch (ClassNotFoundException 
+                    | IllegalAccessException | IllegalArgumentException | InstantiationException 
+                    | NoSuchMethodException | SecurityException | InvocationTargetException ex) {
+                throw new InvalidConfigException("failed to load configured IdentityManager: " + cname, ex);
+            }
         }
-        try {
-            c = Class.forName(cname);
-            Object o = c.newInstance();
-            IdentityManager ret = (IdentityManager) o;
-            log.debug("IdentityManager: " + cname);
-            return ret;
-        } catch (Throwable t) {
-            if (!defaultImplClass.equals(cname) || c != null) {
-                log.error("failed to load configured IdentityManager: " + cname, t);
+        // default
+        return new NoOpIdentityManager();
```

Comment:
Maybe add a few log.debug() lines here to indicate which identity manager is being used?

Reply:
will do

Reply:
improved